File: src/leetcode/python/easy/SumOfLeftLeaves_404.py
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class SumOfLeftLeaves_404(object):
    def sumOfLeftLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        def dfs(node, left):
            if not node:
                return 0
            if not node.left and not node.right and left:
                return node.val
            return dfs(node.left, True) + dfs(node.right, False)

        if not root or not root.left and not root.right:
            return 0
        return dfs(root.left, True) + dfs(root.right, False)

File: src/leetcode/java/easy/SumOfLeftLeaves_404.java
package leetcode.java.easy;

public class SumOfLeftLeaves_404 {
  public int sumOfLeftLeaves(TreeNode root) {
    if (root == null || root.left == null && root.right == null) return 0;
    return dfs(root.left, true) + dfs(root.right, false);
  }

  private int dfs(TreeNode node, boolean left) {
    if (node == null) return 0;
    if (node.left == null && node.right == null && left) {
      return node.val;
    }
    return dfs(node.left, true) + dfs(node.right, false);
  }

  class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int x) { val = x; }
  }
}
